### ISSUE TITLE
Geant4, debug builds and floating point exceptions

### ISSUE THREAD

**Issue by klenze** [FairRootGroup/FairSoft#196]
In the jun19 release, Geant4 enabled floating point exceptions for debug builds. 

Like virtually every instance of a mere library installing an exception handler without user consent (looking at you, ROOT), this is actively harmful. IEEE 754 specified special values (inf, NaN) are legitimate values used in other libraries. Crying "exception" every time such a value is encountered breaks other stuff, anecdotally ROOT histogram painting. 

While perhaps being a minority of one in the FairRoot userbase on this, I really like using debug builds. Something about fewer "??" lines in backtraces and getting code listings instead of asm listings in gdb. :-)

In the pre-spacks release, I fixed this by adding 
`sed s/-DG4FPE_DEBUG//g -i cmake/Modules/Geant4MakeRules_cxx.cmake`
in the relevant build script. 

Recent versions of Geant4 have fixed this (preventing me from bitterly complaining about it to them), instead adding special DEBUG_FPE builds for that. 

In conclusion, if the next version of Geant4 you ship with FairSoft still exhibits this behavior, I would kindly ask you to include a fix similar to my sed script in the relase. 

Thanks,
   Philipp

**Comment by dennisklein** [comment]
We will include Geant4 10.06 or later in the next FairSoft release. It seems Geant4 has added a special CMake build type `Debug_FPE` which sets the definition which you do not like, but not in the *normal* `Debug` build AFAICS. So we should be fine. Pls re-open/notify us if I am wrong or if you have more insight into what we should care about.
